HumanDynamicsEstimationLibrary/algorithms/include/hde/algorithms/DynamicalInverseKinematics.hpp

```diff
@@ -201,7 +201,7 @@ class hde::algorithms::DynamicalInverseKinematics
 
     void clearProblem();
 
-    bool solve(const double dt);
+    bool solve(const double dt, bool &reset);
```

Comment:
Sorry, I just looked a bit more in the PR and realized:
* that this is a public header, so adding an argument would break downstream users (such as https://github.com/ami-iit/kindynfusion/blob/7fafd5d278570b2500ad654345a68cde4beba710/src/lib/inversekinematics/src/IKHelper.cpp#L418). While that is not particularly problematic in general (this is research code, I think this is unnecessary in this case).
* I looked in the logic, and I am afraid that having a reset argument that is passed by reference and gets set to false after it is confused is quite complex as an API. Perhaps it is easier to pass it as value, and by default set to `false`, and then reset the boolean where the function is used?
```suggestion
    bool solve(const double dt, bool reset=false);
```